global-hooks/discovery/cluster_uuid.go

```diff
@@ -71,13 +73,16 @@ func createConfigMapWithUUID(patch go_hook.PatchCollector, clusterUUID string) {
 // There is CM kube-system/d8-cluster-uuid with cluster uuid. Hook must store it to `global.discovery.clusterUUID`.
 // Or generate uuid and create CM
 func discoveryClusterUUID(input *go_hook.HookInput) error {
-	uuidSnap := input.Snapshots["cluster_uuid"]
-
 	const valPath = "global.discovery.clusterUUID"
 
+	uuidSnap, err := sdkobjectpatch.UnmarshalToStruct[string](input.NewSnapshots, "cluster_uuid")
+	if err != nil {
+		return err
```

Comment:
add error verbosity to returned values
```suggestion
		return fmt.Errorf(""failed to unmarshal cluster_uuid snapshot: %w", err)
```